modules/bday.js

```diff
@@ -38,216 +50,245 @@ const confirmChannel = (response_url, cmd, channel) => (
-        channel,
-        text: [
-          `:tada: Birthday Alert for ${bdayPersonFullName} :tada:`,
-          `*${bdayPersonFullName}'s* birthday is coming up soon! Take some time and leave a nice message for ${bdayPersonFullName} to read. Thanks! :smile:`,
-          `Click :point_right: <${MIRO_URL}|here> :point_left: to sign! Instructions are found inside the card.`
-        ].join('\n'),
-        blocks: [
-          {
-            'type': 'header',
-            'text': {
-              'type': 'plain_text',
-              'text': `:tada: Birthday Alert for ${bdayPersonFullName} :tada:`,
-              'emoji': true
-            }
+
+    // when more fields are added
+    if (type === 'block_actions' && action.block_id === 'manage_fields') {
+      let updatedBlocks
+      let _buttons
+
+      if (action.action_id === 'add') {
+        _buttons = blocks.pop()
+        if (blocks.length === 3) {
+          _buttons.elements.push(removeButton)
+        }
+        updatedBlocks = [...blocks, ..._blocks(ref + 1), _buttons]
+      } else {
+        _buttons = blocks.pop()
+        blocks.splice(-3)
+        if (blocks.length === 3) {
+          updatedBlocks = [...blocks, button]
+        } else {
+          updatedBlocks = [...blocks, _buttons]
+        }
+      }
+      const private_metadata = JSON.stringify({ ref: ref + 1, response_url, command, channel_id })
+      return web.views.update({
+        view_id,
+        hash,
+        view: {
+          private_metadata,
+          'callback_id': 'bday',
+          'type': 'modal',
+          'title': {
+            'type': 'plain_text',
+            'text': 'Bday details'
           },
-          {
-            'type': 'section',
-            'text': {
-              'type': 'mrkdwn',
-              'text':  `*${bdayPersonFullName}*'s birthday is coming up soon! Take some time and leave a nice message for ${bdayPersonFullName} to read. Thanks! :smile:`
-            }
+          'blocks': updatedBlocks,
+          'close': {
+            'type': 'plain_text',
+            'text': 'Nevermind'
           },
-          {
-            'type': 'section',
-            'text': {
-              'type': 'mrkdwn',
-              'text': `Click :point_right: <${MIRO_URL}|here> :point_left: to sign! Instructions are found inside the card.`
-            }
-          }
-        ]
-      }))
-    )
-      .then(() => {
+          'submit': {
+            'type': 'plain_text',
+            'text': 'Send'
+          },
+        },
+      })
+    }
+
+    // upon modal submission
+    if (type === 'view_submission') {
+      // send to every member in #general excluding bday person
+      const { members } = await web.conversations.members({ channel: channel_id })
+
+      for (let [user_index, { id }] of Object.entries(data)) {
+        members.splice(members.findIndex((m) => m === id), 1)
+        const { user: { real_name } } = await web.users.info({ user: id })
```

Comment:
js follows camelCase convention, would be getter to assign as `readName`

Reply:
it is from slack response https://api.slack.com/methods/users.info